Weather AlertTests/DataControllerTests.swift

```diff
@@ -31,4 +33,101 @@ class DataControllerTests: XCTestCase {
+            
+            XCTAssertNotNil(object.id, "id shouldnt be nil")
+            XCTAssertTrue(object.id.intValue == 2643743, "id is incorrect")
+            
+            XCTAssertNotNil(object.name, "name shouldnt be nil")
+            XCTAssertEqual(object.name, "London", "name is incorrect")
+            
+            XCTAssertNotNil(object.country, "country shouldnt be nil")
+            XCTAssertEqual(object.country, "GB", "country is incorrect")
+            
+            XCTAssertNotNil(object.lon, "lon shouldnt be nil")
+            XCTAssertEqual(object.lon, -0.13, "lon is incorrect")
+            
+            XCTAssertNotNil(object.lat, "lat shouldnt be nil")
+            XCTAssertEqual(object.lat, 51.51, "lat is incorrect")
+            
+            XCTAssertNotNil(object.deg, "deg shouldnt be nil")
+            XCTAssertEqual(object.deg, 290, "deg is incorrect")
+            
+            XCTAssertNotNil(object.speed, "speed shouldnt be nil")
+            XCTAssertEqual(object.speed,  1.5, "speed is incorrect")
+            
+            
+            
+            exp.fulfill()
+            
+        }
+        
+        waitForExpectations(timeout: 10.0) { (err) in
+            XCTAssertNil(err)
+        }
+    }
+    
+    func testMapSeveralCity() {
+        
+        let exp = expectation(description: "JSON converted to object successfully")
+        
+        DispatchQueue.global().async {
+            
+            let jsonResult = self.loadJson(jsonFile: "SeveralCities")
+            XCTAssertNotNil(jsonResult, "Failed to load file")
+            
+            let list = jsonResult!["list"] as? [[String : Any]]
+            XCTAssertNotNil(list, "List of cities cannot be extracted")
+            
+            let count = jsonResult!["cnt"] as? Int
+            XCTAssertNotNil(count, "Count cannot be extracted")
+            
+            XCTAssertTrue(list!.count == count!, "List elements count doesnt match count variable")
+            
+            let idsToCompare: [NSNumber] = [524901, 703448, 2643743]
+            
+            for (idx, json) in list!.enumerated() {
+                
+                let entity: NSEntityDescription? = NSEntityDescription.entity(forEntityName: "City", in: self.dataController.managedObjectContext)
+                XCTAssertNotNil(entity, "Entity could not be found")
+                
+                let object = City(entity: entity!, insertInto: self.dataController.managedObjectContext)
+                
+                object.mapping(json: json)
+                
+                XCTAssertNotNil(object.id, "id shouldnt be nil")
+                XCTAssertEqual(object.id, idsToCompare[idx], "id shouldnt be nil")
+            }
+            
+            exp.fulfill()
+            
+        }
+        
+        waitForExpectations(timeout: 10.0) { (err) in
+            XCTAssertNil(err)
+        }
+    }
+    
+    func loadJson(jsonFile: String) -> [String : Any]? {
```

Comment:
@AlexKel Baby steps. Let the man figure out how to refactor code and improve it.
Initially, the load and conversion of the file was done inside the first test function.
When he added another test, the implementation moved to a separate function to avoid rewriting code. Once he creates a new test file, I expect him to see he cannot use this function and move if to a different, helper class. Give him a chance.